src/components/icon-search-popover/index.js

```diff
@@ -150,6 +151,11 @@ const IconSearchPopover = props => {
 			fr.onload = function( e ) {
 				setIsDropping( false )
 				const svgString = cleanSvgString( addCustomIconClass( e.target.result ) )
+
+				if ( isPro && props.showPrompt ) {
+					applyFilters( 'stackable.global-settings.inspector.icon-library.prompt', null, svgString )
+				}
+
```

Comment:
Instead of creating a hook specifically for the icon library, it would be best to create a hook for when an icon is loaded in general. Also, use `doAction` here since we are not doing anything to the output value. 

Note the change to `doAction` and hook name:
```suggestion
				doAction( 'stackable.icon-search-popover.svg-upload', svgString )

```

The checking if `isPro` can be done in the premium side:

```js
// Somewhere in the premium code:
if ( isPro ) {
	addAction( 'stackable.icon-search-popover.svg-upload', svgString => {
		// ...
	} )
}
```